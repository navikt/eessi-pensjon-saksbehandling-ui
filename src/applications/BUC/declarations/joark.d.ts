
export interface JoarkFileVariant {
  variantformat:string;
  filnavn: string;
}

export interface JoarkDoc {
  dokumentInfoId: string;
  tittel: string;
  dokumentvarianter: Array<JoarkFileVariant>;
}
export interface JoarkPoster {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: string;
  tilleggsopplysninger?: string;
  dokumenter: Array<JoarkDoc>
}

export interface JoarkFile {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: Date;
  dokumentInfoId: string;
  tilleggsopplysninger?: string;
  variant: JoarkFileVariant
}

interface Content {
  base64: string;
}

export interface JoarkFileWithContent extends JoarkFile {
  content: Content;
  name: string;
  size: number;
  mimetype: string;
}
