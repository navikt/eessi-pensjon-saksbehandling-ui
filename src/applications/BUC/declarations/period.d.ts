export interface PeriodDate {
  day: string;
  month: string;
  year: string;
}

export interface Country {
  label: string;
  value: string;
}

export interface Period {
  id?: number,
  type: string;
  startDate: PeriodDate;
  endDate?: PeriodDate;
  dateType: string;
  uncertainDate: boolean;
  country: Country;
  comment: string;
  otherType?: string;
  insuranceName?: string;
  insuranceType?: string;
  insuranceId ?: string;
  workActivity ?: string;
  workName ?: string;
  workType ?: string;
  workStreet ?: string;
  workCity ?: string;
  workZipCode ?: string;
  workRegion ?: string;
  childFirstName ?: string;
  childLastName ?: string;
  childBirthDate ?: PeriodDate;
  learnInstitution ?: string;
  payingInstitution ?: string;
  attachments ?: Array[any];
  [key: string]: any;
}

export type Periods = Array<Period>

export type PeriodErrors = {[k: string]: string | undefined}

export interface PayloadPeriod {
  land: string;
  periode: null | {
    lukketPeriode?: {
      fom: string | null;
      tom: string | null;
    }
    openPeriode?:  {
      fom: string | null;
      extra: string;
    }
  };
  vedlegg ?: Array[any];
  trygdeordningnavn ?: string;
  medlemskap ?: string;
  forsikkringEllerRegistreringNr ?: string;
  annenInformasjon ?: string;
  usikkerDatoIndikator ?: string;
  jobbUnderAnsattEllerSelvstendig ?: string;
  navnFirma ?: string;
  typePeriode ?: string;
  adresseFirma ?: {
    postnummer: string | undefined;
    by: string | undefined;
    land: string;
    gate: string | undefined;
    region: string | undefined;
  };
  navnPaaInstitusjon ?: string;
  informasjonBarn ?:  {
    etternavn: string | null | undefined;
    fornavn: string | null | undefined;
    foedseldato: string | null | undefined;
    land: string;
  }
}

export interface Payload {
  ansattSelvstendigPerioder ?: Array<PayloadPeriod>;
  boPerioder ?: Array<PayloadPeriod>;
  barnepassPerioder ?: Array<PayloadPeriod>;
  frivilligPerioder ?: Array<PayloadPeriod>;
  forsvartjenestePerioder ?: Array<PayloadPeriod>;
  foedselspermisjonPerioder ?: Array<PayloadPeriod>;
  opplaeringPerioder ?: Array<PayloadPeriod>;
  arbeidsledigPerioder ?: Array<PayloadPeriod>;
  sykePerioder ?: Array<PayloadPeriod>;
  andrePerioder ?: Array<PayloadPeriod>;
}

export interface P4000Info {
  person: any,
  bank: any,
  stayAbroad: Array<Period>
}

export interface P4000PayloadInfo {
  personInfo: any,
  bankInfo: any,
  periodeInfo: Array<PayloadPeriod>
}
