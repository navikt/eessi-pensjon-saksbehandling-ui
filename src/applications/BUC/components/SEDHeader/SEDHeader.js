import React from 'react'
import PT from 'prop-types'
import { Nav } from 'eessi-pensjon-ui'

import './SEDHeader.css'

const SEDHeader = (props) => {
  const { t } = props
  return (
    <div
      className='a-buc-c-sedheader pb-1'
      id='a-buc-c-sedheader__div-id'
    >
      <div className='a-buc-c-sedheader__head col-2'>
        <Nav.Element>{t('buc:form-name')}</Nav.Element>
      </div>
      <div className='a-buc-c-sedheader__head col-4'>
        <Nav.Element>{t('buc:form-status')}</Nav.Element>
      </div>
      <div className='a-buc-c-sedheader__head col-4'>
        <Nav.Element>{t('buc:form-receiver')}</Nav.Element>
      </div>
      <div className='a-buc-c-sedheader__head col-2' />
    </div>
  )
}

SEDHeader.propTypes = {
  t: PT.func.isRequired
}

export default SEDHeader
