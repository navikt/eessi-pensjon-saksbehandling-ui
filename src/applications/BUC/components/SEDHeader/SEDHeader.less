.a-buc-c-sedheader {

  transform: translateX(-20px);
  opacity: 0;
  animation: a-buc-c-sedheader-slideInFromLeft 0.2s forwards;

  &__border {
    &-full {
      border: 1px solid #59514B;
      border-radius: 4px;
    }
    &-bottom {
      border-bottom: 1px solid #59514B;
    }
  }

  &__content {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__column {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  &__status {
    flex-direction: row;
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }

  &__institution {
    flex-direction: row;
    display: flex;
    align-items: center;
  }

  &__actions {
    flex-direction: row;
    align-items: center;
  }
}

@keyframes a-buc-c-sedheader-slideInFromLeft {
  0% {
    opacity: 0;
    transform: translateX(-20px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
}
