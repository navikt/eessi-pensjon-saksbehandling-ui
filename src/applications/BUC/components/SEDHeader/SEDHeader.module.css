@import '../../../../components/Animations/animations.modules.css';

.sedHeader {
  width: 100%;
  padding: 0;
  transform: translateX(-20px);
  opacity: 0;
  animation: slideInFromLeft 0.2s forwards;
  border: none;
  background: transparent !important;
}

.expandDiv {
  margin-left: auto;
}

.attachmentsDiv {
  margin-right: 0.5rem;
}
