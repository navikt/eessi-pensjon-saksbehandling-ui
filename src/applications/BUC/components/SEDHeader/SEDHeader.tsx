import { getBucTypeLabel } from 'applications/BUC/components/BUCUtils/BUCUtils'
import InstitutionList from 'applications/BUC/components/InstitutionList/InstitutionList'
import SEDStatus from 'applications/BUC/components/SEDStatus/SEDStatus'
import FilledPaperClipIcon from 'assets/icons/filled-version-paperclip-2'
import { HorizontalSeparatorDiv } from 'components/StyledComponents'
import { Buc, Institutions, Participant, Sed, Seds } from 'declarations/buc'
import { BucPropType, SedPropType, SedsPropType } from 'declarations/buc.pt'
import { State } from 'declarations/reducers'
import { AllowedLocaleString } from 'declarations/types'
import _ from 'lodash'
import { buttonLogger } from 'metrics/loggers'
import moment from 'moment'
import { Flatknapp } from 'nav-frontend-knapper'
import Panel from 'nav-frontend-paneler'
import { Element, Normaltekst } from 'nav-frontend-typografi'
import { theme, themeHighContrast } from 'nav-styled-component-theme'
import PT from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { useSelector } from 'react-redux'
import styled, { ThemeProvider } from 'styled-components'

export interface SEDHeaderProps {
  buc: Buc;
  className ?: string;
  followUpSeds: Seds;
  onSEDNew: (buc: Buc, sed: Sed) => void;
  sed: Sed;
  style?: React.CSSProperties;
}

export interface SEDHeaderSelector {
  highContrast: boolean
  locale: AllowedLocaleString
}

const mapState = (state: State): SEDHeaderSelector => ({
  highContrast: state.ui.highContrast,
  locale: state.ui.locale
})

const SEDHeaderPanel = styled(Panel)`
  width: 100%;
  padding: 0rem;
  transform: translateX(-20px);
  opacity: 0;
  animation: slideInFromLeft 0.2s forwards;
  border-bottom: ${({theme}: any) => theme.type === 'themeHighContrast' ?
  `2px solid $theme['main-disabled-color']` :
  `1px solid $theme['navGra60']`};
`
const SEDHeaderContent = styled.div`
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
`
const SEDNameDiv = styled.div`
  display: flex;
  flex: 4;
  flex-direction: column;
  justify-content: center;
`
const SEDStatusDiv = styled.div`
  display: flex;
  flex-direction: row;
  flex: 3;
  align-items: flex-start;
  justify-content: flex-start;
  flex-wrap: wrap;
`
const SEDStatusItemDiv= styled.div`
  display: flex;
  align-items: flex-start;
  margin-bottom: 0.5rem;
  margin-right: 0.5rem;
`
const SEDInstitutionsDiv = styled.div`
  flex: 3;
  flex-direction: row;
  display: flex;
  align-items: center;
`
const SEDActionsDiv = styled.div`
  flex: 2;
  flex-direction: row;
  display: flex;
  align-items: center;
  justify-content: flex-start;
`
const SEDAttachmentsDiv = styled.div``

const SEDHeader: React.FC<SEDHeaderProps> = ({
  buc, className, followUpSeds, onSEDNew, sed, style
}: SEDHeaderProps): JSX.Element => {
  const institutionList: Institutions = sed.participants
    ? sed.participants.filter((participant) => {
      return (sed.status === 'received') ? participant.role === 'Sender'
        : (sed.status !== 'draft') ? participant.role !== 'Sender'
          : participant.organisation.countryCode === 'NO'
    })
      .map((participant: Participant) => ({
        country: participant.organisation.countryCode,
        institution: participant.organisation.name
      })) : []
  const { highContrast, locale }: SEDHeaderSelector = useSelector<State, SEDHeaderSelector>(mapState)
  const { t } = useTranslation()
  const sedLabel: string = getBucTypeLabel({
    t: t,
    type: sed.type,
    locale: locale
  })

  return (
    <ThemeProvider theme={highContrast ? themeHighContrast: theme}>
      <SEDHeaderPanel
        style={style}
        className={className}
      >
        <SEDHeaderContent>
          <SEDNameDiv>
            <Element>
              {sed.type}{sedLabel ? ' - ' + sedLabel : ''}
            </Element>
          </SEDNameDiv>
          <SEDStatusDiv>
            <SEDStatusItemDiv>
              <SEDStatus highContrast={highContrast} status={sed.status} />
              <HorizontalSeparatorDiv date-size='0.5'/>
              <Normaltekst
                data-testId='a-buc-c-sedheader__lastUpdate'
                data-tip={t('ui:lastUpdate')}
              >
                {sed.lastUpdate && moment(sed.lastUpdate).format('DD.MM.YYYY')}
              </Normaltekst>
              {sed.version && (
                <Normaltekst
                  data-testId='a-buc-c-sedheader__version'
                >
                  {t('ui:version')}{': '}{sed.version || '-'}
                </Normaltekst>
              )}
            </SEDStatusItemDiv>
            {sed.version !== '1' && (
              <SEDStatusItemDiv>
                <SEDStatus highContrast={highContrast} status={'first_' + sed.status} />
                <HorizontalSeparatorDiv date-size='0.5'/>
                <Normaltekst
                  data-id='a-buc-c-sedheader__firstSend'
                  data-tip={t('ui:status-first')}
                >
                  {sed.firstVersion ? moment(sed.firstVersion.date).format('DD.MM.YYYY') : null}
                </Normaltekst>
              </SEDStatusItemDiv>
            )}
          </SEDStatusDiv>
          <SEDInstitutionsDiv>
            <InstitutionList
              locale={locale}
              type='separated'
              institutions={institutionList}
            />
          </SEDInstitutionsDiv>
          <SEDActionsDiv>
            {!_.isEmpty(sed.attachments) && (
              <SEDAttachmentsDiv
                data-testId='a-buc-c-sedheader__actions-attachments'
                data-tip={t('buc:form-youHaveXAttachmentsInSed',
                  { attachments: sed.attachments.length })}
              >
                <FilledPaperClipIcon />
              </SEDAttachmentsDiv>
            )}
            {(!_.isEmpty(followUpSeds) && sed.status === 'received') && (
              <Flatknapp
                mini
                data-amplitude='buc.list.besvarSed'
                data-testId='a-buc-c-sedheader__actions-answer-button'
                onClick={(e: React.MouseEvent) => {
                  buttonLogger(e)
                  onSEDNew(buc, sed)
                }}
              >
                {t('buc:form-answerSED')}
              </Flatknapp>
            )}
          </SEDActionsDiv>
        </SEDHeaderContent>
      </SEDHeaderPanel>
    </ThemeProvider>
  )
}

SEDHeader.propTypes = {
  buc: BucPropType.isRequired,
  className: PT.string,
  followUpSeds: SedsPropType.isRequired,
  onSEDNew: PT.func.isRequired,
  sed: SedPropType.isRequired,
  style: PT.object
}

export default SEDHeader
