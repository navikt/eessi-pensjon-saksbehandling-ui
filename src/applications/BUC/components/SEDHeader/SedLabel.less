.a-buc-sed-sedlabel {
  div.panel {
      background: #e9e9e9;
  }
  div.c-ui-sed-name{
      width:100px;
  }
  div.c-ui-status {
      width:200px;
  }
}
