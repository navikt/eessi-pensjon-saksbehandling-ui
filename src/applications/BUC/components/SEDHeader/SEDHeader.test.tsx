import { Buc, Sed } from 'declarations/buc'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import mockBucs from 'mocks/buc/bucs'
import { stageSelector } from 'setupTests'
import SEDHeader, { SEDHeaderProps, SEDHeaderSelector } from './SEDHeader'

const defaultSelector: SEDHeaderSelector = {
  locale: 'nb'
}

describe('applications/BUC/components/SEDHeader/SEDHeader', () => {
  const buc: Buc = mockBucs()[0]
  const sed: Sed = buc.seds![0]
  sed.status = 'received'
  const initialMockProps: SEDHeaderProps = {
    onSEDNew: jest.fn(),
    buc: buc,
    sed: sed,
    followUpSeds: [buc.seds![1]]
  }
  let wrapper: ReactWrapper

  beforeAll(() => {
    console.log('SEDHeader')
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<SEDHeader {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.a-buc-c-sedheader')).toBeTruthy()
    expect(wrapper.find('.a-buc-c-sedheader__name').hostNodes().render().text()).toEqual('X008 - buc:buc-X008')

    const status = wrapper.find('.a-buc-c-sedheader__status').hostNodes()
    expect(status.find('SEDStatus').render().text()).toEqual('buc:status-' + sed.status)
    expect(status.find('Normaltekst.a-buc-c-sedheader__lastUpdate').render().text()).toEqual('23.10.2019')

    const institutions = wrapper.find('.a-buc-c-sedheader__institutions').hostNodes()
    expect(institutions.find('InstitutionList').render().text()).toEqual('NAV ACCEPTANCE TEST 07')

    const actions = wrapper.find('.a-buc-c-sedheader__actions').hostNodes()
    expect(actions.exists('Icons')).toBeTruthy()
    expect(actions.exists('Flatknapp.a-buc-c-sedheader__actions-answer-button')).toBeTruthy()
    const replySedButton = wrapper.find('.a-buc-c-sedheader__actions-answer-button').hostNodes().first()
    replySedButton.simulate('click')
    expect(initialMockProps.onSEDNew).toBeCalledWith(buc, sed)
  })
})
