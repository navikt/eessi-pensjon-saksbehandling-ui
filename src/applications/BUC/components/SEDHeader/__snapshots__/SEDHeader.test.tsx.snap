// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/SEDHeader/SEDHeader Render: match snapshot 1`] = `
.c27 {
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  display: inline-block;
  line-height: 0px;
  margin: 0.25rem;
}

.c27.animate {
  opacity: 0.01;
  -webkit-animation: iPdXhn 0.5s forwards;
  animation: iPdXhn 0.5s forwards;
}

.c27.type-original.wave div {
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-animation: eRATHR 4s ease-in-out infinite;
  animation: eRATHR 4s ease-in-out infinite;
}

.c27.type-circle.wave div img {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-animation: eRATHR 4s ease-in-out infinite;
  animation: eRATHR 4s ease-in-out infinite;
}

ui-fss.g1[id="c26"] {
  content: "c27,";
}

.c29.type-circle {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 1px solid var(--navds-semantic-color-border);
  border-radius: 50%;
  overflow: hidden;
  position: relative;
}

.c29.type-circle img {
  position: absolute;
  margin: auto;
  top: -999px;
  left: -999px;
  bottom: -999px;
  right: -999px;
}

.c29.type-circle.size-XS {
  width: 16px;
  height: 16px;
  min-width: 16px;
  min-height: 16px;
  max-width: 16px;
  max-height: 16px;
}

.c29.type-circle.size-XS img {
  height: 18px;
}

.c29.type-circle.size-S {
  width: 24px;
  height: 24px;
  min-width: 24px;
  min-height: 24px;
  max-width: 24px;
  max-height: 24px;
}

.c29.type-circle.size-S img {
  height: 26px;
}

.c29.type-circle.size-M {
  width: 32px;
  height: 32px;
  min-width: 32px;
  min-height: 32px;
  max-width: 32px;
  max-height: 32px;
}

.c29.type-circle.size-M img {
  height: 35px;
}

.c29.type-circle.size-L {
  width: 40px;
  height: 40px;
  min-width: 40px;
  min-height: 40px;
  max-width: 40px;
  max-height: 40px;
}

.c29.type-circle.size-L img {
  height: 45px;
}

.c29.type-circle.size-XL {
  width: 48px;
  height: 48px;
  min-width: 48px;
  min-height: 48px;
  max-width: 48px;
  max-height: 48px;
}

.c29.type-circle.size-XL img {
  height: 54px;
}

.c29.type-circle.size-XXL {
  width: 72px;
  height: 72px;
  min-width: 72px;
  min-height: 72px;
  max-width: 72px;
  max-height: 72px;
}

.c29.type-circle.size-XXL img {
  height: 82px;
}

.c29.type-original.size-XS img {
  height: 16px;
  min-height: 16px;
  width: auto;
}

.c29.type-original.size-S img {
  height: 24px;
  min-height: 24px;
  width: auto;
}

.c29.type-original.size-M img {
  height: 32px;
  min-height: 32px;
  width: auto;
}

.c29.type-original.size-L img {
  height: 40px;
  min-height: 40px;
  width: auto;
}

.c29.type-original.size-XL img {
  height: 48px;
  min-height: 48px;
  width: auto;
}

.c29.type-original.size-XXL img {
  height: 72px;
  min-height: 72px;
  width: auto;
}

ui-fss.g2[id="c28"] {
  content: "c29,";
}

.c25 {
  padding: 0.25rem;
}

ui-fss.g5[id="c24"] {
  content: "c25,";
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c23:not(.noMargin) {
  margin-bottom: 0.35rem;
}

ui-fss.g6[id="c22"] {
  content: "c23,";
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g7[id="c20"] {
  content: "c21,";
}

.c31 {
  margin-left: 0.5rem;
}

.c31.oneLine {
  white-space: nowrap;
}

ui-fss.g8[id="c30"] {
  content: "c31,";
}

.c13.received {
  background-color: var(--navds-global-color-lightblue-500) !important;
}

.c13.active {
  background-color: var(--navds-global-color-purple-200) !important;
  border-color: var(--navds-global-color-purple-500) !important;
}

.c13.first_sent,
.c13.first_received,
.c13.first_cancelled {
  background-color: var(--navds-global-color-gray-400) !important;
}

ui-fss.g9[id="c12"] {
  content: "c13,";
}

.c15 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g27[id="c14"] {
  content: "c15,";
}

.c37 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g40[id="c36"] {
  content: "c37,";
}

.c9 {
  margin: 0.3rem;
  color: var(--navds-semantic-color-text);
}

ui-fss.g59[id="c8"] {
  content: "c9,";
}

.c11 {
  display: inline-block;
}

ui-fss.g60[id="c10"] {
  content: "c11,";
}

.c33 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

ui-fss.g76[id="c32"] {
  content: "c33,";
}

.c35 {
  margin-right: 0.5rem;
}

ui-fss.g77[id="c34"] {
  content: "c35,";
}

.c3 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

ui-fss.g78[id="c2"] {
  content: "c3,";
}

.c1 {
  width: 100%;
  padding: 0rem;
  -webkit-transform: translateX(-20px);
  -ms-transform: translateX(-20px);
  transform: translateX(-20px);
  opacity: 0;
  -webkit-animation: fSLHgq 0.2s forwards;
  animation: fSLHgq 0.2s forwards;
  border: none;
  background: transparent !important;
}

ui-fss.g79[id="c0"] {
  content: "c1,";
}

.c19 {
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0.3rem;
}

ui-fss.g80[id="c18"] {
  content: "c19,";
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 4;
  -ms-flex: 4;
  flex: 4;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

ui-fss.g81[id="c4"] {
  content: "c5,";
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 0.5rem;
  margin-right: 0.5rem;
  margin-top: 0.5rem;
}

ui-fss.g82[id="c6"] {
  content: "c7,";
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

ui-fss.g83[id="c16"] {
  content: "c17,";
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c29.type-original {
    display: inline-block;
  }

  .c29.type-original img {
    border: 1px solid var(--navds-semantic-color-border);
  }
}

@supports not (-ms-high-contrast:none) {
  .c29.type-original {
    border: 1px solid var(--navds-semantic-color-border);
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
  }
}

<div
  class="navds-panel c0 c1"
>
  <div
    class="c2 c3"
  >
    <div
      class="c4 c5"
    >
      <p
        class="navds-detail"
        data-testid="a_buc_c_sedheader--name-id"
      >
        P2000
         - buc:buc-P2000
      </p>
      <div
        class="c6 c7"
      >
        <div
          aria-hidden="true"
          class="navds-popover navds-popover--hidden"
          data-placement="right"
          style="position: absolute; top: 0px; left: 0px;"
          tabindex="-1"
        >
          <div
            class="c8 c9"
          >
            <p
              class="navds-body-long"
            >
              ui:firstVersion
              29.05.2019
            </p>
          </div>
          <div
            class="navds-popover__arrow"
            style="left: -0.5rem;"
          />
        </div>
        <div
          class="c10 c11"
          id="386338e6-0957-4423-83fb-552544e8e27d"
        >
          <span
            class="navds-tag c12 c13 received navds-tag--info navds-tag--medium navds-body-short"
            data-testid="sedstatus"
          >
            buc:status-received
          </span>
        </div>
        <div
          class="c14 c15"
        />
        <div
          class="c16 c17"
        >
          <p
            class="navds-body-long"
            data-testid="a_buc_c_sedheader--version-date-id"
          >
            29.05.2019
          </p>
          <p
            class="navds-body-long"
            data-testid="a_buc_c_sedheader--version-id"
          >
            ui:version
            : 
            5
          </p>
        </div>
      </div>
    </div>
    <div
      class="c18 c19"
    >
      <div
        class="c20 c21"
        data-testid="a_buc_c_sedheader--institutions-id"
      >
        <div
          class="c22 c23"
          data-testid="a_buc_c_institutionlist--div-id"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="top"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <span
              class="c24 c25"
            >
              Norge
            </span>
            <div
              class="navds-popover__arrow"
              style="bottom: -0.5rem;"
            />
          </div>
          <div
            id="flag-a3c4bb30-6a42-4957-97bc-650cdf4e7b50"
            style="display: inline-block;"
          >
            <div
              class="c26 c27 size-S type-circle animate"
            >
              <div
                class="c28 c29 size-S type-circle"
              >
                <img
                  alt="Norge"
                  src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1100 800'%3e %3crect width='1100' height='800' fill='%23ef2b2d'/%3e %3crect width='200' height='800' x='300' fill='%23fff'/%3e %3crect width='1100' height='200' y='300' fill='%23fff'/%3e %3crect width='100' height='800' x='350' fill='%23002868'/%3e %3crect width='1100' height='100' y='350' fill='%23002868'/%3e %3c/svg%3e"
                />
              </div>
            </div>
          </div>
          <p
            class="c30 c31 navds-body-long"
          >
            NO:DEMONO02
          </p>
        </div>
      </div>
    </div>
    <div
      class="c18 c19"
    >
      <div
        class="c20 c21"
        data-testid="a_buc_c_sedheader--institutions-id"
      >
        <div
          class="c22 c23"
          data-testid="a_buc_c_institutionlist--div-id"
        >
          <div
            aria-hidden="true"
            class="navds-popover navds-popover--hidden"
            data-placement="top"
            style="position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <span
              class="c24 c25"
            >
              Norge
            </span>
            <div
              class="navds-popover__arrow"
              style="bottom: -0.5rem;"
            />
          </div>
          <div
            id="flag-99beb983-9026-4fbb-afc2-e2cd2c083550"
            style="display: inline-block;"
          >
            <div
              class="c26 c27 size-S type-circle animate"
            >
              <div
                class="c28 c29 size-S type-circle"
              >
                <img
                  alt="Norge"
                  src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1100 800'%3e %3crect width='1100' height='800' fill='%23ef2b2d'/%3e %3crect width='200' height='800' x='300' fill='%23fff'/%3e %3crect width='1100' height='200' y='300' fill='%23fff'/%3e %3crect width='100' height='800' x='350' fill='%23002868'/%3e %3crect width='1100' height='100' y='350' fill='%23002868'/%3e %3c/svg%3e"
                />
              </div>
            </div>
          </div>
          <p
            class="c30 c31 navds-body-long"
          >
            NO:DEMONO01
          </p>
        </div>
      </div>
    </div>
    <div
      class="c32 c33"
      data-testid="a_buc_c_sedheader--actions-id"
    >
      <div
        aria-hidden="true"
        class="navds-popover navds-popover--hidden"
        data-placement="right"
        style="position: absolute; top: 0px; left: 0px;"
        tabindex="-1"
      >
        <div
          class="c8 c9"
        >
          <span>
            buc:form-youHaveXAttachmentsInSed
          </span>
        </div>
        <div
          class="navds-popover__arrow"
          style="left: -0.5rem;"
        />
      </div>
      <div
        class="c10 c11"
        id="599cf83f-2031-445d-9989-efb507eb88f6"
      >
        <div
          class="c34 c35"
          data-testid="a_buc_c_sedheader--actions-attachments"
        >
          <svg
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M2 0a2 2 0 0 0-2 2v20a2 2 0 0 0 2 2h20a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2Zm11.377 3.307a4.312 4.312 0 0 1 6.185 0 4.49 4.49 0 0 1 0 6.258l-6.925 7.076a3.07 3.07 0 0 1-4.401 0 3.18 3.18 0 0 1 0-4.436l4.267-4.36a1 1 0 1 1 1.43 1.398l-4.268 4.361a1.18 1.18 0 0 0 0 1.638 1.07 1.07 0 0 0 1.543 0l6.924-7.076a2.49 2.49 0 0 0 0-3.46 2.312 2.312 0 0 0-3.326 0L6.07 13.634c-1.425 1.457-1.425 3.827 0 5.283a3.554 3.554 0 0 0 5.109 0l.006-.007 2.059-2.063c.016-.019.032-.037.05-.054l1.499-1.5 5.5-5.5a1 1 0 1 1 1.415 1.414l-5.499 5.5-3.604 3.612a5.554 5.554 0 0 1-7.965-.003c-2.185-2.234-2.185-5.847 0-8.08l8.738-8.93Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        data-amplitude="buc.view.besvarSed"
        data-testid="a_buc_c_sedheader--answer-button-id"
      >
        <span
          aria-live="polite"
          class="navds-label"
        >
          buc:form-answerSED
        </span>
      </button>
      <div
        class="c36 c37"
      />
    </div>
  </div>
</div>
`;
